runtime/src/iree/base/internal/atomics.h

```diff
@@ -134,35 +134,68 @@ extern "C" {
 // operands without caring about the exact bit sizes at each site.
 
 typedef iree_atomic_int32_t iree_atomic_ref_count_t;
-#define iree_atomic_ref_count_init(count_ptr) \
-  iree_atomic_store_int32(count_ptr, 1, iree_memory_order_relaxed)
+
+// Why atomic_init and not a relaxed atomic store: this would only avoid
+// undefined behavior but we would still get incorrect initialization if there
+// was a race, so we still want to avoid races. Avoiding undefined behavior
+// only means that TSan couldn't catch such a race.
+#define iree_atomic_ref_count_init_value(count_ptr, value) \
```

Comment:
at the point there's a non-1 initial value, maybe we shouldn't be using ref count atomics and instead use iree_atomic_int32? if all of the semantics for ordering match except for this then it's fine, but if we find ourselves needing different iree_atomic_ref_count_* functions for ref counts and other things we'll probably want to reconsider.

Reply:
I want to use the atomic ref_count *functions* in iree_task_scope_t to be piggyback on all the right decisions re memory orders, and assorted rationale comments.

The key is that it really is a stable invariant, that the ref_count *functions* are what I want to use --- because the contract of iree_task_scope_t really is "like an atomic refcount around a thing that posts notification on destruction plus some custom instrumentation".

As I want to use the *functions*, I technically should use the *type* too, as the typedef could theoretically change in the future.